### DISCUSSION TITLE
Passing referer option to curl, but brew audit returns 403

### DISCUSSION THREAD

**Discussion by cesarcoatl** [Homebrew/discussions#576] [Tap maintenance and Homebrew development] [answered]
I have created a tap and added a formula that in order to download the file from the `url`, I have to pass the `referer` option.

```ruby
  url "https://files.foo.com/release/foo/1.0/foo-1.0.zip", referer: "https://foo.com/"
```

So `curl` is executed with the option `-e https://foo.com/`.

The formula downloads and installs correctly, but running `brew audit` results in the following error:

```bash
$ brew audit --online --git --skip-style foo@1.0
foo@1.0:
  * Stable: The URL https://files.foo.com/release/foo/1.0/foo-1.0.zip is not reachable (HTTP status code 403)
Error: 1 problem in 1 formula detected
```

The tap I created using `brew tap-new` installed two workflows, and my PRs are failing at the `brew test-bot --only-formulae` step.

If I'm using the default download strategy, `curl`, and it is able to download the file without issues, why doesn't `brew audit` also use the `referer` option?

Thanks

**Comment by cesarcoatl** [comment]
Anyone?

**Comment by Rylan12** [comment, marked as answer]
If it's just `brew audit` that's failing, you can probably still merge. By the time `brew audit` runs, the bottles should already have been built and they should be uploaded regardless of the `brew audit` status.

**Reply by Rylan12** [reply]
A better solution, though, is probably to fix the audit to be able to use the `-e` option as well (although tbh I'm not 100% sure what happens under the hood there). If I have time tomorrow, I'll try to play with the audit to see if it can be coaxed into understanding that.

You're welcome, of course, to take a look and make the change yourself if desired. I can't promise a quick turnaround on this as I'm pretty busy.

**Reply by Rylan12** [reply]
Do you happen to know of an example of a situation where the referer is needed? It just makes it a tad easier to test if you have a specific case in mind. If not, I'll do my best

**Reply by cesarcoatl** [reply]
Thanks, @Rylan12.

This is the formula I created: [ignition@8.1.1](https://github.com/coatl-dev/homebrew-coatl-dev/blob/coatl/Formula/ignition@8.1.1.rb)

**Reply by Rylan12** [reply]
Thanks, that's very helpful! I'll take a look tomorrow.

**Reply by Rylan12** [reply]
I've opened https://github.com/Homebrew/brew/pull/10402 to modify `brew audit` so that it sees the specs passed to the `url` method. That should solve the issue.

**Reply by cesarcoatl** [reply]
That's awesome! Thanks, @Rylan12.

I assume once that's merged the work flow running on my repo will use the latest version.

**Reply by Rylan12** [reply]
Correct
